### PR TITLE
Ddcnl 17

### PR DESCRIPTION
**Pull request by Dandyboyo2**
Converted acceptnce tests' content tests to unit tests within this frontend.

### CODE REVIEW

File: test/uk/gov/hmrc/nisp/views/LandingViewSpec.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    override implicit val templateRenderer: TemplateRenderer = MockTemplateRenderer
+
+    override def httpPost: HttpPost = mockHttp
+
+    override def localSubmitUrl(implicit request: Request[AnyContent]): String = ""
+
+    override def contactFormReferer(implicit request: Request[AnyContent]): String = request.headers.get(REFERER).getOrElse("")
+
+    override val applicationConfig: ApplicationConfig = new ApplicationConfig {
```
Comment (danewatts) [inline test/uk/gov/hmrc/nisp/views/LandingViewSpec.scala]: You can use test/uk/gov/hmrc/nisp/utils/FakeApplicationConfig.scala object instead

File: test/uk/gov/hmrc/nisp/views/LandingViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nisp.views
+
+import org.scalatest.mock.MockitoSugar
+import play.api.mvc.{AnyContent, Request}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsString, _}
```
Comment (danewatts) [inline test/uk/gov/hmrc/nisp/views/LandingViewSpec.scala]: In all your tests you have this import. The first is importing just contentAsString, the `_` imports everything. You dont need the first part
```suggestion
import play.api.test.Helpers._
```

Comment (danewatts) [reply test/uk/gov/hmrc/nisp/views/LandingViewSpec.scala]: You should make this change in all of your test files

File: test/uk/gov/hmrc/nisp/views/TimeoutViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      val expected = ">" + messages(messageKey) + "<"
+      title must include(expected)
+    }
+
+    "assert correct href on the start again button" in {
+      val redirect = source.getElementsByTag("a").get(1).attr("href")
+      val expected = "/check-your-state-pension/account"
+      redirect must include(expected)
+      println(redirect)
+      println(expected)
```
Comment (danewatts) [inline test/uk/gov/hmrc/nisp/views/TimeoutViewSpec.scala]: print lines

File: test/uk/gov/hmrc/nisp/views/StatePension_CopeViewSpec.scala [hunk 4]
```diff
@@ -23,7 +23,7 @@ import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.mock.MockitoSugar
 import play.api.i18n.Messages
 import play.api.test.FakeRequest
-import play.api.test.Helpers.{contentAsString, _}
+import play.api.test.Helpers.{_}
```
Comment (liampan) [inline test/uk/gov/hmrc/nisp/views/StatePension_CopeViewSpec.scala:26]:
```suggestion
import play.api.test.Helpers._
```

File: test/uk/gov/hmrc/nisp/views/NotAuthorisedViewSpec.scala [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      val title = source.title()
+      val expected = messages("nisp.iv.failure.general.title") + Constants.titleSplitter +
+                     messages("nisp.title.extension") + Constants.titleSplitter + messages("nisp.gov-uk")
+      title must include(expected)
+    }
+
+    "assert correct paragraph text on page" in {
+      val title = source.getElementsByTag("p").get(1).toString
+      val messageKey = "nisp.iv.failure.general.youcan"
+      val expected = ">" + messages(messageKey) + "<"
```
Comment (Tateni93) [inline test/uk/gov/hmrc/nisp/views/NotAuthorisedViewSpec.scala:77]: Does this need to be a concatenated String. Can the assertion not be made directly on the message.

File: test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      val title = source.title()
+      val expected = messages("nisp.tandcs.title") + Constants.titleSplitter +
+                     messages("nisp.title.extension") + Constants.titleSplitter + messages("nisp.gov-uk")
+      title must include(expected)
+    }
+
+    "assert correct heading title on page" in {
+      val title = source.getElementsByTag("h1").get(0).toString
+      val messageKey = "nisp.tandcs.title"
+      val expected = "> " + messages(messageKey) + " <"
```
Comment (Tateni93) [inline test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala:77]: Does this need to be constructed like this?

File: test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      title must include(expected)
+    }
+
+    "assert correct heading level 2 on page" in {
+      val title = source.getElementsByTag("h2").get(0).toString
+      val messageKey = "nisp.tandcs.heading"
+      val expected = ">" + messages(messageKey) + "<"
```
Comment (Tateni93) [inline test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala:84]: Does this need to be constructed like this?

File: test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      title must include(expected)
+    }
```
Comment (Tateni93) [inline test/uk/gov/hmrc/nisp/views/TermsAndConditionsViewSpec.scala:86]: Given that there is a h1 and h2 on this page. Is there any other content that could also be captured in this spec?

**Comment by lewey** [comment]
Closed as work appears under a newer branch

**Pull request closed without merging**